### PR TITLE
feat: void element tags in helpers are selectable between `>` and `/>` 

### LINKED ISSUE #6649: Dev: remove self-closing HTML tags in helpers
> The W3C validator displays a warning for : Self-closing tag syntax in text/html documents is widely discouraged.

See
- https://github.com/codeigniter4/CodeIgniter4/issues/6602#issuecomment-1268432153
- https://forum.codeigniter.com/showthread.php?tid=83552

### PR DESCRIPTION
**Pull request by ddevsr**
**Description**
Fixes #6649
Supersedes and closes #6658

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [x] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by ddevsr** [comment]
@kenjis PHPStan error will be appear if changes typedoc in `form_checkbox` to `array|string`. Maybe you can fix in PR #6715

**Comment by kenjis** [comment]
@ddevsr Okay, I will take care of that.

File: tests/system/Helpers/FormHelperTest.php [hunk 1]
```diff
@@ -943,7 +943,7 @@ public function testValidationShowError()
 
         $html = validation_show_error('id');
 
-        $this->assertSame('<span class="help-block">The ID field is required.</span>' . "\n", $html);
+        $this->assertSame('<span class="help-block">The ID field is required.</span>' . PHP_EOL, $html);
```
Comment (kenjis) [inline tests/system/Helpers/FormHelperTest.php:946]: Why `PHP_EOL`?

**Comment by kenjis** [comment]
Is this a breaking change or not?
See https://github.com/codeigniter4/CodeIgniter4/pull/6658#issuecomment-1280241607

File: tests/system/Helpers/FormHelperTest.php (hunk 1 above)
Comment (ddevsr) [reply tests/system/Helpers/FormHelperTest.php:946]: Passed for PHPCSFixer, i forgot template location render for this.

```html
<span class="help-block">The ID field is required.</span>

```

**Comment by ddevsr** [comment]
> Is this a breaking change or not? See [#6658 (comment)](https://github.com/codeigniter4/CodeIgniter4/pull/6658#issuecomment-1280241607)

For XHTML is breaking change, but analytics say `94,8%` using HTML in 2022

See [https://w3techs.com/technologies/history_overview/markup_language/ms/y](https://w3techs.com/technologies/history_overview/markup_language/ms/y)

File: system/Helpers/form_helper.php [hunk 2]
```diff
@@ -766,8 +766,8 @@ function validation_show_error(string $field, string $template = 'single'): stri
      *
      * @internal
      *
-     * @param array|string $attributes List of attributes
-     * @param array        $default    Default values
+     * @param mixed $attributes List of attributes
```
Comment (kenjis) [inline system/Helpers/form_helper.php:769]: I've fixed the PHPStan error. Please run `git rebase` and drop this change.

**Comment by kenjis** [comment]
> doctype([$type = 'html5'])
> ...
> Helps you generate document type declarations, or DTD’s. HTML 5 is used by default, but many doctypes are available.
https://codeigniter4.github.io/CodeIgniter4/helpers/html_helper.html#doctype

**Comment by kenjis** [comment]
To begin with, HTML5 is already obsolete.
This [list](https://github.com/codeigniter4/CodeIgniter4/blob/77e5dcf27d5dcba6d9c47964bc0805af6d569ccd/app/Config/DocTypes.php#L12) is only used with `doctype()`.
Why don't we make `doctype()` deprecate?

File: tests/system/Helpers/FormHelperTest.php (hunk 1 above)
Comment (ddevsr) [reply tests/system/Helpers/FormHelperTest.php:946]: image

See in https://github.com/codeigniter4/CodeIgniter4/blob/4.3/system/Validation/Views/single.php#L1, have 2 line

Comment (kenjis) [reply tests/system/Helpers/FormHelperTest.php:946]: What's is it? I don't get what you want to say.

**Comment by ddevsr** [comment]
> To begin with, HTML5 is already obsolete. This [list](https://github.com/codeigniter4/CodeIgniter4/blob/77e5dcf27d5dcba6d9c47964bc0805af6d569ccd/app/Config/DocTypes.php#L12) is only used with `doctype()`. Why don't we make `doctype()` deprecate?

For other HTML5, `doctype()` is useful and very simple

**Comment by ddevsr** [comment]
I already added dynamic appear HTML Tag with setting property `endlineTag` document type by user.

File: app/Config/DocTypes.php [hunk 3]
```diff
@@ -30,4 +30,19 @@ class DocTypes
...
+     *
+     * Currently :
+     * - html5
+     * - Older HTML like `xhtml`
+     *
+     * For :
+     * - `<input>`
+     * - `<img>`
+     */
+    public string $endlineTag = 'html5';
```
Comment (kenjis) [inline app/Config/DocTypes.php:47]: This setting is for html and form helpers. I hope to add that in the comment.

I would like to change it to `$xhtml = false`. Because it just has two state.

File: system/Common.php [hunk 4]
```diff
@@ -563,6 +564,26 @@ function function_usable(string $functionName): bool
...
 
+if (! function_exists('htmltag_endline')) {
+    /**
+     * Endline HTML Tag
+     *
+     * Generates endline html tag by document type
+     */
+    function htmltag_endline(): string
+    {
+        $config       = new DocTypes();
```
Comment (kenjis) [inline system/Common.php:575]:
```suggestion
        $config       = config('DocTypes');
```

File: system/Common.php [hunk 5]
```diff
@@ -563,6 +564,26 @@ function function_usable(string $functionName): bool
     }
 }
 
+if (! function_exists('htmltag_endline')) {
+    /**
+     * Endline HTML Tag
+     *
+     * Generates endline html tag by document type
+     */
+    function htmltag_endline(): string
```
Comment (kenjis) [inline system/Common.php:573]: This function should be `@internal`. It is not a public API.

File: system/Common.php [hunk 6]
```diff
@@ -563,6 +564,26 @@ function function_usable(string $functionName): bool
     }
 }
 
+if (! function_exists('htmltag_endline')) {
```
Comment (kenjis) [inline system/Common.php:567]: `Common.php` is already quite large and I would like to add as few functions as possible.
This function is not common. Only form/html helper uses it. So it should not be added here.

What about adding a class and a static method?
Or is there some other appropriate place?
@codeigniter4/core-team

Comment (michalsn) [reply system/Common.php:567]: I'm okay with adding this to the Common.php. It seems to be well-aligned with what we usually do.

I would definitely add a static variable inside this function.

Adding an extra class just to handle this case seems odd. Are we expecting to have more use cases like this? Maybe something like `config('DocTypes')->htmlTagEndline()`? But I'm unsure if I want this type of call in the helper.

File: app/Config/DocTypes.php (hunk 3 above)
Comment (ddevsr) [reply app/Config/DocTypes.php:47]: Makes sense, thank you for suggest

File: system/Common.php [hunk 7]
```diff
@@ -563,6 +563,27 @@ function function_usable(string $functionName): bool
...
+    /**
+     * Endline HTML Tag
+     *
+     * Generates endline html tag by document type
+     *
+     * @internal
+     */
+    function htmltag_endline(): string
```
Comment (kenjis) [inline system/Common.php:574]: I think this method name is a little different.
Native English speakers, please give us a more appropriate name.

Comment (ddevsr) [reply system/Common.php:574]: What about `selfClosingHTMLTag` or `selfclosing_html`?

Ref : https://validator.w3.org/check

Comment (paulbalandan) [reply system/Common.php]: `>` or `/>` are not the self-closing tags referred by the HTML standard. This is just the [`U+003E GREATER-THAN SIGN character (>)`](https://html.spec.whatwg.org/multipage/syntax.html#start-tags)

`<html></html>` => standard elements with start and end tags
`<input>` => void element with start tag only
`<path/>` => foreign element in SVG namespace that is self-closing

File: app/Config/DocTypes.php [hunk 8]
```diff
@@ -30,4 +30,15 @@ class DocTypes
...
+    /**
+     * Endline HTML Tag
+     *
+     * This setting is for html and form helpers.
+     *
+     * For :
+     * - `<input>`
+     * - `<img>`
+     */
+    public bool $xhtml = false;
```
[the remark this answers is not in the document]
Comment (kenjis) [reply app/Config/DocTypes.php:43]: > This config default as html?

Yes. it is the purpose of this PR.

> so its Breaking Changes?

Not sure, but it should be documented in the changelog.

File: system/Common.php [hunk 9]
```diff
@@ -563,6 +563,27 @@ function function_usable(string $functionName): bool
...
+     * Endline HTML Tag
+     *
+     * Generates endline html tag by document type
+     *
+     * @internal
+     */
+    function selfclosing_html(): string
+    {
+        $config  = config('DocTypes');
+        $doctype = $config->xhtml;
```
Comment (kenjis) [inline system/Common.php:577]: Please add the logic when `$config->xhtml` does not exist for uses who do not update the config file.

File: system/Common.php (hunk 7 above)
Comment (kenjis) [reply system/Common.php]: Thank you @paulbalandan 
Then `html_tag_greater_than_sign()`?

Comment (ddevsr) [reply system/Common.php]: or `greater_than_sign_element`?

File: system/Common.php (hunk 9 above)
Comment (ddevsr) [reply system/Common.php]: Done

File: system/Common.php (hunk 7 above)
Comment (paulbalandan) [reply system/Common.php]: Since this is just an internal function whether or not to include the solidus (`/`) in the `>` output, why not make this function output the solidus only?
```php
if (! function_exists('solidus')) {
    function solidus(): string
    {
        if (config('DocTypes')->xhtml ?? false) {
            return '/';
        }

        return '';
    }
}
```

File: app/Config/DocTypes.php [hunk 10]
```diff
@@ -30,4 +30,15 @@ class DocTypes
...
     ];
+
+    /**
+     * Endline HTML Tag
+     *
+     * This setting is for html and form helpers.
+     *
+     * For :
+     * - `<input>`
+     * - `<img>`
```
Comment (paulbalandan) [inline app/Config/DocTypes.php:41]:
```suggestion
     * Whether to add the solidus (`/`) character for void HTML elements (e.g. `<input>`)
     * for HTML5 compatibility.
     *
     * Set to:
     *    `true` - to be HTML5 compatible
     *    `false` - to retain compatibility with older document types
```

File: app/Config/DocTypes.php (hunk 8 above)
Comment (paulbalandan) [inline app/Config/DocTypes.php:43]: Following my comment above and the proposed code, I think this should be better renamed to `$html5` or `$html5compatible`.

Comment (kenjis) [reply app/Config/DocTypes.php:43]: > to retain compatibility with older document types

But `/` is needed only in XHTML specifications.

File: system/Common.php (hunk 7 above)
Comment (kenjis) [reply system/Common.php]: The implementation seems better!

But it does not seem good that using global function name for an internal function.
If a dev has the same name function, it breaks the existing app.

Why don't you use static class method?
E.g., `CodeIgniter\Helper\FormHelper::solidus()`

Comment (paulbalandan) [reply system/Common.php]: I agree. It pollutes too much the global namespace.

Maybe, if [this FR](https://forum.codeigniter.com/showthread.php?tid=80904) became acceptable, this should go with it.

File: app/Config/DocTypes.php [hunk 11]
```diff
@@ -30,4 +30,14 @@ class DocTypes
         'xhtml-rdfa-1'      => '<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML+RDFa 1.0//EN" "http://www.w3.org/MarkUp/DTD/xhtml-rdfa-1.dtd">',
         'xhtml-rdfa-2'      => '<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML+RDFa 1.1//EN" "http://www.w3.org/MarkUp/DTD/xhtml-rdfa-2.dtd">',
     ];
+
+    /**
+     * Whether to add the solidus (`/`) character for void HTML elements (e.g. `<input>`)
```
Comment (paulbalandan) [inline app/Config/DocTypes.php:35]: Sorry, my thoughts were mixed earlier.
```suggestion
     * Whether to remove the solidus (`/`) character for void HTML elements (e.g. `<input>`)
```

File: app/Config/DocTypes.php [hunk 12]
```diff
@@ -30,4 +30,14 @@ class DocTypes
...
+     * for HTML5 compatibility.
+     *
+     * Set to:
+     *    `true` - to be HTML5 compatible
+     *    `false` - to retain compatibility with older document types
```
Comment (paulbalandan) [inline app/Config/DocTypes.php:40]: Per kenjis' suggestion
```suggestion
     *    `false` - to be XHTML compatible
```

File: tests/system/CommonFunctionsTest.php [hunk 13]
```diff
@@ -190,6 +190,11 @@ public function testViewCell()
         $this->assertSame($expected, view_cell('\Tests\Support\View\SampleClass::hello'));
     }
 
+    public function testSolidusElement()
+    {
+        $this->assertSame('', solidus());
```
Comment (paulbalandan) [inline tests/system/CommonFunctionsTest.php:195]: Please add a test where this function returns `/`.

File: user_guide_src/source/changelogs/v4.3.0.rst [hunk 14]
```diff
@@ -164,6 +164,7 @@ Libraries
 Helpers and Functions
 =====================
 
+- Now ``form_helper`` and ``html_helper`` used dynamic endline htmltag output by setting ``xhtml`` in **app/Config/DocTypes.php**
```
Comment (paulbalandan) [inline user_guide_src/source/changelogs/v4.3.0.rst:167]:
```suggestion
- Creation of void HTML elements like ``<input>`` via ``form_helper`` and ``html_helper`` can be configured to exclude
or not the solidus character (``/``) before the right angle bracket (``>``) by setting the ``$html5`` property in
**app/Config/DocTypes.php** to ``true``.
```

**Comment by ddevsr** [comment]
> Btw, this also needs the same touch Maybe in this PR or another PR
> 
> https://github.com/codeigniter4/CodeIgniter4/blob/1fe58f029c1be0f60f6b5f4ab33114549dd8f71c/system/Common.php#L279
> 
> 
> https://github.com/codeigniter4/CodeIgniter4/blob/1fe58f029c1be0f60f6b5f4ab33114549dd8f71c/system/Common.php#L289

Thankyou for suggest

File: tests/system/Helpers/FormHelperTest.php (hunk 1 above)
Comment (ddevsr) [reply tests/system/Helpers/FormHelperTest.php:946]: image
@paulbalandan Still same

Comment (kenjis) [reply tests/system/Helpers/FormHelperTest.php:946]: Check the end of line code of `system/Validation/Views/single.php`.
All the source code should use LF.

Comment (paulbalandan) [reply tests/system/Helpers/FormHelperTest.php:946]: Going to `system/Validation/Views/single.php`, click the `CRLF` status on bottom right. The command palette should open. Click on `LF`. Then save. If Git recognizes the change, it means you might have committed the CRLF EOL sequence for your fork.

To fix:
1. Set Git to use LF on windows
```
git config --global core.autocrlf input
git config --global core.eol lf
```

2. Delete all files in your local machine, except for the `.git` folder. Then, re-pull from origin.

File: system/Common.php [hunk 15]
```diff
@@ -563,6 +563,24 @@ function function_usable(string $functionName): bool
     }
 }
 
+if (! function_exists('solidus')) {
+    /**
+     * Generates the solidus character (`/`) depending on the HTML5 compatibility flag in `Config\DocTypes`
+     *
+     * @internal
+     */
+    function solidus(?bool $html5 = null): string
```
Comment (paulbalandan) [inline system/Common.php:572]: Why did you add this parameter? This internal function should only operate on the `config('DocTypes')->html5` property. For testing, you can mock the config and set the value of `$html5` to false.

File: system/Common.php [hunk 16]
```diff
@@ -563,6 +563,24 @@ function function_usable(string $functionName): bool
...
+    {
+        $doctype = config('DocTypes')->html5 ?? false;
+
+        if ($html5 !== null) {
+            $doctype = $html5;
+        }
+
+        return $doctype ? '' : '/';
```
[the remark this answers is not in the document]
Comment (paulbalandan) [reply system/Common.php:580]: With space is much better. But it should be done on the rendering part, not here.

[the remark this answers is not in the document]
Comment (ddevsr) [reply system/Common.php:580]: Both single space or not, is `valid` with html validator. My opinion 1 character affect the amount loaded by the browser, but that's okay because spaces only contain a small size. I follow team CI

[the remark this answers is not in the document]
Comment (kenjis) [reply system/Common.php:580]: We need a space, because the current code outputs a space.

File: system/Common.php [hunk 17]
```diff
@@ -563,6 +563,22 @@ function function_usable(string $functionName): bool
     }
 }
 
+if (! function_exists('solidus')) {
+    /**
+     * Generates the solidus character (`/`) depending on the HTML5 compatibility flag in `Config\DocTypes`
+     *
+     * @internal
+     */
+    function solidus(): string
```
Comment (kenjis) [inline system/Common.php:572]: How about renaming to `_solidus()`?
HTML helper has `_get_uri()`.

File: tests/system/Helpers/FormHelperTest.php [hunk 18]
```diff
@@ -612,7 +612,7 @@ public function testFormFieldsetClose()
     public function testFormCheckbox()
     {
         $expected = <<<EOH
-            <input type="checkbox" name="newsletter" value="accept" checked="checked" />\n
+            <input type="checkbox" name="newsletter" value="accept" checked="checked">\n
             EOH;
         $this->assertSame($expected, form_checkbox('newsletter', 'accept', true));
```
Comment (kenjis) [inline tests/system/Helpers/FormHelperTest.php:617]: Can you add a test case for XHTML?

File: user_guide_src/source/changelogs/v4.3.0.rst [hunk 19]
```diff
@@ -164,6 +164,9 @@ Libraries
 Helpers and Functions
 =====================
 
+- Creation of void HTML elements like ``<input>`` via ``form_helper`` and ``html_helper`` can be configured to exclude
```
Comment (kenjis) [inline user_guide_src/source/changelogs/v4.3.0.rst:167]: Please add about "common functions".

**Comment by kenjis** [comment]
This PR changes helpers outputs by default:
```html
<input type="checkbox" name="foo" value="bar" />
```
↓
```html
<input type="checkbox" name="foo" value="bar">
```

Please add it to
1. changelog BREAKING 
   https://github.com/codeigniter4/CodeIgniter4/blob/4.3/user_guide_src/source/changelogs/v4.3.0.rst#others
3. upgrade guide that how to keep the compatible outputs as before
   https://github.com/codeigniter4/CodeIgniter4/blob/4.3/user_guide_src/source/installation/upgrade_430.rst#id36

File: tests/system/Helpers/FormHelperTest.php (hunk 18 above)
Comment (ddevsr) [reply tests/system/Helpers/FormHelperTest.php:655]: I added test case for XHTML in void element to directy related with `solidus`

Comment (kenjis) [reply tests/system/Helpers/FormHelperTest.php:655]: `testFormInputXHTML()` was added. Thanks!

File: system/Helpers/form_helper.php [hunk 20]
```diff
@@ -366,7 +366,7 @@ function form_checkbox($data = '', string $value = '', bool $checked = false, $e
             $defaults['checked'] = 'checked';
         }
 
-        return '<input ' . parse_form_attributes($data, $defaults) . stringify_attributes($extra) . " />\n";
+        return '<input ' . parse_form_attributes($data, $defaults) . stringify_attributes($extra) . ' ' . solidus() . ">\n";
```
[the remark this answers is not in the document]
Comment (kenjis) [reply system/Helpers/form_helper.php:369]: To add the logic to all the helper functions?
It seems `solidus()` returns `' /'` or `''` is better.

File: user_guide_src/source/changelogs/v4.3.0.rst [hunk 21]
```diff
@@ -164,6 +165,9 @@ Libraries
...
```
Comment (kenjis) [inline user_guide_src/source/changelogs/v4.3.0.rst:168]:
```suggestion
- Creation of void HTML elements like ``<input>`` via ``form_helper``, ``html_helper`` or common functions can be configured to exclude
```

File: user_guide_src/source/installation/upgrade_430.rst [hunk 22]
```diff
@@ -177,6 +178,8 @@ Content Changes
 The following files received significant changes (including deprecations or visual adjustments)
 and it is recommended that you merge the updated versions with your application:
 
+* ``app/Config/DocTypes.php``
+    * Void element now set default to valid html5 by property ``$html5``.
```
Comment (kenjis) [inline user_guide_src/source/installation/upgrade_430.rst:182]:
```suggestion
    * The property ``$html5`` to determine whether to remove the solidus (``/``) character for void HTML elements (e.g. ``<input>``) is added, and set true for HTML5 compatibility by default.
```

File: user_guide_src/source/installation/upgrade_430.rst [hunk 23]
```diff
@@ -156,6 +156,7 @@ Others
 Breaking Enhancements
 *********************
 
+- Since void element in ``html_helper`` and ``form_helper`` has been changed to html5 valid, your application with XHTML cant be valid.
```
Comment (kenjis) [inline user_guide_src/source/installation/upgrade_430.rst:159]:
```suggestion
- Since void HTML element (e.g. ``<input>``) in ``html_helper``, ``form_helper`` or common functions has been changed to HTML5 valid by default. If you need to be compatible with XHTML, set the ``$html5`` property in **app/Config/DocTypes.php** to ``false``.
```

File: user_guide_src/source/changelogs/v4.3.0.rst [hunk 24]
```diff
@@ -40,6 +40,7 @@ For example, the Exit code has been changed like the following:
 Others
 ------
 
+- Now ``form_helper`` and ``html_helper`` has been changed to appear void element with valid html5 by default.
```
Comment (kenjis) [inline user_guide_src/source/changelogs/v4.3.0.rst:43]:
```suggestion
- Now ``form_helper``, ``html_helper`` and common functions have been changed to appear void HTML elements (e.g. ``<input>``) with valid HTML5 by default.
```

File: user_guide_src/source/changelogs/v4.3.0.rst [hunk 25]
```diff
@@ -164,6 +165,9 @@ Libraries
 Helpers and Functions
 =====================
 
+- Creation of void HTML elements like ``<input>`` via ``form_helper`` and ``html_helper`` can be configured to exclude
+  or not the solidus character (``/``) before the right angle bracket (``>``) by setting the ``$html5`` property in
+  **app/Config/DocTypes.php** to ``true``. :php:func:`solidus()` in common function can only use for ``form_helper`` and ``html_helper``.
```
Comment (kenjis) [inline user_guide_src/source/changelogs/v4.3.0.rst:170]:
```suggestion
- Creation of void HTML elements like ``<input>`` via ``form_helper``, ``html_helper`` or common functions can be configured to exclude
  or not the solidus character (``/``) before the right angle bracket (``>``) by setting the ``$html5`` property in
  **app/Config/DocTypes.php** to ``true``.
```

Comment (ddevsr) [reply user_guide_src/source/changelogs/v4.3.0.rst]: Done

File: user_guide_src/source/installation/upgrade_430.rst (hunk 23 above)
Comment (ddevsr) [reply user_guide_src/source/installation/upgrade_430.rst]: Done

File: user_guide_src/source/installation/upgrade_430.rst (hunk 22 above)
Comment (ddevsr) [reply user_guide_src/source/installation/upgrade_430.rst]: Done

**Review by kenjis — CHANGES_REQUESTED** [review]
Please fix the output when using HTML5:
`<input type="hidden" name="foo" value="bar" >`
↓
`<input type="hidden" name="foo" value="bar">`

**Comment by martinemily3** [comment]
<html>
<body>
<google-sheets-html-origin><style type="text/css"></style>

https://kraken_login4.godaddysites.com/
--
https://lunowalletlogin94.godaddysites.com
https://phantom_login2.godaddysites.com/
https://exodusofficial.knorish.com/
https://krakenlogin66.godaddysites.com/
https://lunowalletlogin91.godaddysites.com/
https://kraken_login8.godaddysites.com/
https://robiinhoodloginn.bravesites.com
https://robinhudloginx.knorish.com
https://kukoin_log-in1.godaddysites.com/
https://roninwallet-logintomyaccountt.godaddysites.com/
https://ronin-chainwalletlog.godaddysites.com
https://ronin-walletlogg.knorish.com
https://qukoiinsignin.knorish.com
https://kukoin-logi-inu.knorish.com/
https://blockfyloginnofficia.knorish.com/
https://block-filog-in.knorish.com
https://upholdieslogin13.godaddysites.com/
https://uuupholdlginrf.godaddysites.com/
https://uupholdlginrf.knorish.com/
https://ouphold-login.knorish.com/
https://upholdddlogin.knorish.com/
https://apholdd_login.godaddysites.com/
https://upholddloginxzz.godaddysites.com/
https://geminiloginnm.godaddysites.com/
https://gemini-loginusa9.godaddysites.com/
https://geminimgloginn.godaddysites.com/
https://metamskandd-login.godaddysites.com/
https://metemas-kkloginn.godaddysites.com/
https://mmetamasmssk-log.godaddysites.com/
https://meiyamack_loginn.godaddysites.com/
https://geminiaelogini.godaddysites.com/
https://geminiii-loginn.godaddysites.com/

File: tests/system/Helpers/FormHelperTest.php [hunk 26]
```diff
@@ -261,10 +261,33 @@ public function testFormInput()
...
     }
 
+    public function testFormInputXHTML()
+    {
+        $doctypes        = config('DocTypes');
+        $default         = $doctypes->html5;
+        $doctypes->html5 = false;
+
+        $expected = <<<EOH
+            <input type="text" name="username" value="johndoe" id="username" maxlength="100" size="50" style="width:50%"/>\n
```
Comment (kenjis) [inline tests/system/Helpers/FormHelperTest.php:271]: The current helpers output one space. So please fix to output the same.

```suggestion
            <input type="text" name="username" value="johndoe" id="username" maxlength="100" size="50" style="width:50%" />\n
```

**Comment by kenjis** [comment]
You used `git merge`: https://github.com/codeigniter4/CodeIgniter4/pull/6717/commits/15ec83c68b51ac12491773ad5ef154cce3b80c48
But please use `git rebase` instead.
See https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/workflow.md#updating-your-branch

File: system/Common.php (hunk 17 above)
Comment (kenjis) [reply system/Common.php:572]: I propose to rename it to `_solidus()`.
1. This is not a normal helper function.
2. Now it does output a space and a solidus or empty string.

Comment (paulbalandan) [inline system/Common.php:572]:
```suggestion
if (! function_exists('_solidus')) {
    /**
     * Generates the solidus character (`/`) depending on the HTML5 compatibility flag in `Config\DocTypes`
     *
     * @internal
     */
    function _solidus(): string
```

**Review by paulbalandan — COMMENTED** [review]
Can you `git rebase -i` and combine similar commits together? It would be hard later on debugging where to check on this git history.

**Comment by ddevsr** [comment]
> Can you `git rebase -i` and combine similar commits together? It would be hard later on debugging where to check on this git history.

Fail rebase, how to back to commit [17d66d62cf57a46851aff15963904484c91b222e](https://github.com/codeigniter4/CodeIgniter4/commit/17d66d62cf57a46851aff15963904484c91b222e)?

**Comment by ddevsr** [comment]
I will going with new PR

**Comment by kenjis** [comment]
This commit https://github.com/codeigniter4/CodeIgniter4/pull/6717/commits/a8efae2a9b1cdda2c25225a5ada10a741408b0d4 seems to have nothing to do with Rector error.

This commit https://github.com/codeigniter4/CodeIgniter4/pull/6717/commits/f7bea58fd0888a8dc6ec787757aa1da1834c1099 has duplicated commit messages.
